pkg/solana/logpoller/filters.go

```diff
@@ -228,7 +231,7 @@ func (fl *filters) removeFilterFromIndexes(filter Filter) {
 		}
 	}
 
-	discriminatorHead := filter.Discriminator()[:10]
+	discriminatorHead := filter.DiscriminatorRawBytes()
 	if refcount, ok := fl.knownDiscriminators[discriminatorHead]; ok {
```

Comment:
I only named this var `discriminatorHead` to distinguish it from a full discriminator, since it was just the first 10 characters of 12. Suggest dropping the `Head` now that it's the full discriminator
```suggestion
	discriminator := filter.DiscriminatorRawBytes()
	if refcount, ok := fl.knownDiscriminators[discriminator]; ok {
```